src/main/java/org/jenkinsci/backend/ircbot/hosting/GradleVerifier.java

```diff
@@ -0,0 +1,427 @@
+import org.kohsuke.github.GitHub;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.HashSet;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+import static org.jenkinsci.backend.ircbot.HostingChecker.LOWEST_JENKINS_VERSION;
+
+public class GradleVerifier extends CodeVisitorSupport implements BuildSystemVerifier {
+
+    public static final String SPECIFY_LICENSE = "Please specify a license in your build.gradle file using the `licenses` closure. See https://github.com/jenkinsci/gradle-jpi-plugin#configuration for an example.";
+    public static final String MISSING_JENKINS_VERSION = "Your build.gradle file is missing the `jenkinsVersion` item.";
+    public static final String MISSING_SHORTNAME = "Your build.gradle file is missing the `shortName` item.";
+    public static final String MISSING_DISPLAYNAME = "Your build.gradle file is missing the `displayName` item.";
+    public static final String MISSING_GROUP = "Your build.gradle file is missing the `group` item. Please add one with the value `io.jenkins.plugins`";
+    public static final String MISSING_GROUP_VALUE = "Your build.gradle file has a `group` but the value is empty. Please set the value to `io.jenkins.plugins` or a company/group specific `group`";
+    public static final String JPI_PLUGIN_VERSION_TOO_LOW = "The version of the gradle jpi plugin in your build.gradle (%s) is to low, please update to at least (%s)";
+    public static final String CANNOT_VERIFY_GROUP = "The `group` value from your build.gradle seems to be a variable, we cannot verify the `group` value in this case, please make sure the group is `io.jenkins.plugins` or a company/group specific `group`";
+    public static final String SHOULD_BE_IO_JENKINS_PLUGINS = "The `group` from the build.gradle should be `io.jenkins.plugins` instead of `org.jenkins-ci.plugins`";
+    public static final String NO_BUILD_GRADLE_FOUND = "No build.gradle found in root of project, if you are using a different build system, or this is not a plugin, you can disregard this message";
+    public static final String INVALID_BUILD_GRADLE = "Your build.gradle file did not parse correctly, please verify the content.";
+    public static final String INCORRECT_TARGET_COMPAT_VERSION = "Only targetCompatibility=1.8 is supported for Jenkins plugins.";
+    public static final String CANNOT_VERIFY_TARGET_COMPAT = "The `targetCompatibility` value from your build.gradle seems to be a variable, we cannot verify the `targetCompatibility` in this case, please make sure it is set to `1.8`";
+public static final String CANNOT_VERIFY_JENKINS_VERSION = "The `jenkinsVersion` value from your build.gradle seems to be a variable, we cannot verify the `jenkinsVersion` in this case, please make sure it is set to at least `%s`";

```

Comment:
I assume there is no significance to the indentation difference between this line and the preceding lines.
```suggestion
    public static final String CANNOT_VERIFY_JENKINS_VERSION = "The `jenkinsVersion` value from your build.gradle seems to be a variable, we cannot verify the `jenkinsVersion` in this case, please make sure it is set to at least `%s`";

```